src/de/jost_net/JVerein/gui/control/SpendenbescheinigungControl.java

```diff
@@ -658,21 +662,117 @@ public Part getSpendenbescheinigungList() throws RemoteException
-    while (spendenbescheinigungen.hasNext())
+    if (spbList != null)
     {
-      spbList.addItem(spendenbescheinigungen.next());
+      try
+      {
+        spbList.removeAll();
+        for (Spendenbescheinigung spb : getSpendenbescheinigungen())
+        {
+          spbList.addItem(spb);
+        }
+      }
+      catch (RemoteException e1)
+      {
+        Logger.error("Fehler", e1);
+      }
     }
   }
 
+  @SuppressWarnings("unchecked")
+  private ArrayList<Spendenbescheinigung> getSpendenbescheinigungen() throws RemoteException
+  {
+    final DBService service = Einstellungen.getDBService();
+    ArrayList<Object> bedingungen = new ArrayList<>();
+    and = false;
+    
+    sql = "select spendenbescheinigung.*  from spendenbescheinigung ";
+    sql +=  "left join mitglied on (spendenbescheinigung.mitglied = mitglied.id) ";
+    
+    if (isSuchnameAktiv() && getSuchname().getValue() != null)
+    {
+      String tmpSuchname = (String) getSuchname().getValue();
+      if (tmpSuchname.length() > 0)
+      {
+        addCondition("(lower(zeile2) like ?)");
+        bedingungen.add("%" + tmpSuchname.toLowerCase() + "%");
+      }
+    }
+    if (isMailauswahlAktiv())
+    {
+      int mailauswahl = (Integer) getMailauswahl().getValue();
+      if (mailauswahl == MailAuswertungInput.OHNE)
+      {
+        addCondition("(email is null or length(email) = 0)");
+      }
+      if (mailauswahl == MailAuswertungInput.MIT)
+      {
+        addCondition("(email is  not null and length(email) > 0)");
+      }
+    }
+    if (isDatumvonAktiv() && getDatumvon().getValue() != null)
+    {
+      addCondition("bescheinigungsdatum >= ?");
+      Date d = (Date) getDatumvon().getValue();
+      bedingungen.add(new java.sql.Date(d.getTime()));
+    }
+    if (isDatumbisAktiv() && getDatumbis().getValue() != null)
+    {
+      addCondition("bescheinigungsdatum <= ?");
+      Date d = (Date) getDatumbis().getValue();
+      bedingungen.add(new java.sql.Date(d.getTime()));
+    }
+    if (isEingabedatumvonAktiv() && getEingabedatumvon().getValue() != null)
+    {
+      addCondition("spendedatum >= ?");
+      Date d = (Date) getEingabedatumvon().getValue();
+      bedingungen.add(new java.sql.Date(d.getTime()));
+    }
+    if (isEingabedatumbisAktiv() && getEingabedatumbis().getValue() != null)
+    {
+      addCondition("spendedatum <= ?");
+      Date d = (Date) getEingabedatumbis().getValue();
+      bedingungen.add(new java.sql.Date(d.getTime()));
+    }
+    sql += " ORDER BY bescheinigungsdatum desc ";
+    
+    ResultSetExtractor rs = new ResultSetExtractor()
+    {
+      @Override
+      public Object extract(ResultSet rs) throws RemoteException, SQLException
+      {
+        ArrayList<Spendenbescheinigung> list = new ArrayList<>();
+        while (rs.next())
+        {
+          list.add(
+              (Spendenbescheinigung) service.createObject(Spendenbescheinigung.class, rs.getString(1)));
```

Comment:
Das hatte ich beim ersten Review übersehen. Siehe meine Anmerkungen in https://github.com/openjverein/jverein/pull/233#pullrequestreview-2117634206